src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -56,11 +56,39 @@ var (
 	errEmptyJitterBucketList   = errors.New("empty jitter bucket list")
 )
 
+// InstanceIDType is the instance ID type that defines how the
+// instance ID is constructed, which is then used to lookup the
+// aggregator instance in the placement.
+type InstanceIDType string
+
+const (
+	// HostIDInstanceIDType specifies to just use the host ID
+	// as the instance ID for lookup in the placement.
+	HostIDInstanceIDType InstanceIDType = "host_id"
+	// HostIDPortInstanceIDType specifies to use the host ID
+	// concatenated with the port to be used for lookup
+	// in the placement (legacy, how the instance ID used to be constructed
+	// which imposed the strange requirement that the instance ID
+	// in the topology used to require the port concat'd with the
+	// host ID).
+	HostIDPortInstanceIDType InstanceIDType = "host_id_port"
+)
+
+// InstanceIDConfiguration is the instance ID configuration.
+type InstanceIDConfiguration struct {
+	// Type specifies how to construct the instance ID that is
```

Comment:
nit: maybe best not to use `Type` as the name?

Reply:
Sure thing, will use InstanceIDType.